contracts/two-party-pol-holder/src/contract.rs

```diff
@@ -167,19 +213,19 @@ fn try_claim(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Response, Con
         msg: to_binary(withdraw_liquidity_hook)?,
     };
 
+    let denom_splits = DENOM_SPLITS.load(deps.storage)?;
+    let mut distribution_messages = denom_splits
+        .get_single_receiver_distribution_messages(withdraw_coins, claim_party.router.to_string());
+
     // we submit the withdraw liquidity message followed by transfer of
     // underlying assets to the corresponding router
-    let withdraw_and_forward_msgs = vec![
-        CosmosMsg::Wasm(WasmMsg::Execute {
-            contract_addr: lp_token,
-            msg: to_binary(withdraw_msg)?,
-            funds: vec![],
-        }),
-        CosmosMsg::Bank(BankMsg::Send {
-            to_address: claim_party.clone().router,
-            amount: withdraw_coins,
-        }),
-    ];
+    let mut withdraw_and_forward_msgs = vec![CosmosMsg::Wasm(WasmMsg::Execute {
+        contract_addr: lp_token_addr.to_string(),
+        msg: to_binary(withdraw_msg)?,
+        funds: vec![],
+    })];
+
+    withdraw_and_forward_msgs.append(&mut distribution_messages);
 
     claim_party.allocation = Decimal::zero();
```

Comment:
You do it as part of this message? might have missed it.

Reply:
Yeah, it was there before this pr so it didn't show up for review:

```rust
    claim_party.allocation = Decimal::zero();

    // if other party had not claimed yet, we assign it the full position
    if !counterparty.allocation.is_zero() {
        counterparty.allocation = Decimal::one();
    } else {
        // otherwise both parties claimed everything and we can complete
        CONTRACT_STATE.save(deps.storage, &ContractState::Complete)?;
    }
```